devito/compiler.py

```diff
@@ -500,6 +503,14 @@ class CustomCompiler(Compiler):
     or otherwise default to ``-fopenmp``.
     """
 
+    def __new__(cls, *args, **kwargs):
+        if any(i in environ for i in ['CC', 'CXX', 'CFLAGS', 'LDFLAGS']):
+            obj = Compiler.__new__(cls, *args, **kwargs)
```

Comment:
why not just a super?

Reply:
yes you're right

Reply:
there are some odd rules with new and init that I forget every time. But I tried adding a `print` to `__init__` and between the two `embed()` below it gets called only once

```
            from IPython import embed; embed()
            obj = super().__new__(cls, *args, **kwargs)
            obj.__init__(*args, **kwargs)
            from IPython import embed; embed()
```

Reply:
is it? I'm overriding `__new__` so how is that possible? 
it's not called upon `obj = super().__new__(cls, *args, **kwargs)` (I just tried), and then I'd just be returning `obj`, so...

Reply:
Oh ok, still confised by how these two play together sometimes,